R/tbl-df.r

```diff
@@ -69,7 +69,7 @@ print.tbl_df <- function(x, ..., n = NULL, width = NULL, n_extra = NULL) {
   # Next, subset rows
   if (!missing(i)) {
     if (length(x) == 0) {
-      nr <- length(attr(x, "row.names")[i])
+      nr <- length(seq_len(nr)[i])
```

Comment:
I don't quite understand your request. Even if for original data frame `has_rownames()` is `TRUE` in case of zero columns after the command `x <- .subset(x, j)` `x` becomes `NULL` and the original
`length(attr(x, "row.names")[i])` gives not what is expected.
Now I have an interesting idea. How about just change the order of subsetting in the original code: first subset rows and then columns? Seems like it will also fix the issue. The possible drawback is slightly increased computation time in the case of many columns due to
``map(x, `[`, i)``. What do you think?